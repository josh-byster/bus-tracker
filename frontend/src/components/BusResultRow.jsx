import React, { Component } from 'react';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';
import posed, { PoseGroup } from 'react-pose';
import removeColors from './HelperFunctions';

class BusResultRow extends Component {
  TransitionWrapper = posed.tr({
    enter: {
      opacity: 1,
      // eslint-disable-next-line react/destructuring-assignment
      delay: (100 * (1 - 0.5 ** this.props.elementOrder)) / (1 - 0.5),
      transition: {
        opacity: { ease: 'easeIn', duration: 300 }
      }
    },
    exit: {
      opacity: 0,
      transition: { duration: 0 }
    }
  });

  constructor(props) {
    super(props);
    this.state = {
      isVisible: false
    };
  }

  componentDidMount() {
    this.setState({ isVisible: true });
  }

  computeHMS = expectedDate => {
    const date = new Date(expectedDate.toString());
    let hour = date.getHours() % 12;
    if (hour === 0) {
      hour = 12;
    }
    let minute = date.getMinutes();
    if (minute < 10) {
      minute = `0${minute}`;
    }
    let seconds = date.getSeconds();
    if (seconds < 10) {
      seconds = `0${seconds}`;
    }
    return `${hour}:${minute}:${seconds}`;
  };

  getTRStyle = () => {
    const { info } = this.props;
    return {
      backgroundColor: `#${info.route.route_color}`,
      color: `#${info.route.route_text_color}`
    };
  };

  render() {
    const { isVisible } = this.state;
    const { info, toggleModal } = this.props;
    return (
      <PoseGroup flipMove={false}>
        {isVisible && (
          <this.TransitionWrapper
            key={info.headsign + info.expected_mins}
            style={this.getTRStyle()}
            className="resultRow"
          >
            <td>
              <b>{removeColors(info.headsign)}</b>
            </td>
            <td>
              {info.expected_mins !== 0
                ? `${info.expected_mins}m`
                : 'Arriving Now'}
            </td>
            <td className="no-wrap">{this.computeHMS(info.expected)}</td>
            <td>
              <Button color="success" onClick={() => toggleModal(info)}>
                Location
              </Button>
            </td>
          </this.TransitionWrapper>
        )}
      </PoseGroup>
    );
  }
}

BusResultRow.propTypes = {
  info: PropTypes.object.isRequired,
  toggleModal: PropTypes.func.isRequired,
  elementOrder: PropTypes.number.isRequired
};

export default BusResultRow;
