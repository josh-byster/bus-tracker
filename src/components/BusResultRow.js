import React, { Component } from "react";
import { removeColors } from "./HelperFunctions";
import { Button } from "reactstrap";
import PropTypes from "prop-types";
import posed, { PoseGroup } from "react-pose";

class BusResultRow extends Component {
  TransitionWrapper = posed.tr({
    enter: {
      y: 0,
      opacity: 1,
      delay: 100 * this.props.elementOrder,
      transition: {
        y: { type: "spring", stiffness: 1000, damping: 15 },
        default: { duration: 300 }
      }
    },
    exit: {
      y: 0,
      opacity: 0,
      transition: { duration: 150 }
    }
  });

  constructor(props) {
    super(props);
    this.state = {
      isVisible: false,
      modalOpen: false
    };
  }

  componentDidMount() {
    this.setState({ isVisible: true });
  }
  computeHMS = expected_date => {
    var date = new Date(expected_date.toString());
    var hour = date.getHours() % 12;
    if (hour === 0) {
      hour = 12;
    }
    var minute = date.getMinutes();
    if (minute < 10) {
      minute = "0" + minute;
    }
    var seconds = date.getSeconds();
    if (seconds < 10) {
      seconds = "0" + seconds;
    }
    return `${hour}:${minute}:${seconds}`;
  };

  getTRStyle = () => {
    return {
      backgroundColor: "#" + this.props.info.route.route_color,
      color: "#" + this.props.info.route.route_text_color
    };
  };

  render() {
    return (
      <PoseGroup>
        {this.state.isVisible && (
          <this.TransitionWrapper
            key="bus"
            style={this.getTRStyle()}
            className="resultRow"
          >
            <td>
              <b>{removeColors(this.props.info.headsign)}</b>
            </td>
            <td>
              {this.props.info.expected_mins !== 0
                ? this.props.info.expected_mins + "m"
                : "Arriving Now"}
            </td>
            <td className="no-wrap">
              {this.computeHMS(this.props.info.expected)}
            </td>
            <td>
              <Button
                color="success"
                onClick={e => this.props.toggleModal(this.props.info)}
              >
                Location
              </Button>
            </td>
          </this.TransitionWrapper>
        )}
      </PoseGroup>
    );
  }
}

BusResultRow.propTypes = {
  info: PropTypes.object.isRequired,
  toggleModal: PropTypes.func.isRequired,
  elementOrder: PropTypes.number.isRequired
};

export default BusResultRow;
