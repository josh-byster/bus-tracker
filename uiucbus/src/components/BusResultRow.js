import React, { Component } from "react";
import { removeColors } from "./HelperFunctions";
import { Button } from "reactstrap";
import PropTypes from "prop-types";
class BusResultRow extends Component {
  computeHMS = expected_date => {
    var date = new Date(expected_date.toString());
    var hour = date.getHours() % 12;
    if (hour === 0) {
      hour = 12;
    }
    var minute = date.getMinutes();
    if (minute < 10) {
      minute = "0" + minute;
    }
    var seconds = date.getSeconds();
    if (seconds < 10) {
      seconds = "0" + seconds;
    }
    return `${hour}:${minute}:${seconds}`;
  };

  getTRStyle = () => {
    return {
      backgroundColor: "#" + this.props.info.route.route_color,
      color: "#" + this.props.info.route.route_text_color
    };
  };
  render() {
    return (
      <tr style={this.getTRStyle()}>
        <td>
          <b>{removeColors(this.props.info.headsign)}</b>
        </td>
        <td>
          {this.props.info.expected_mins !== 0
            ? this.props.info.expected_mins + "m"
            : "Arriving Now"}
        </td>
        <td>{this.computeHMS(this.props.info.expected)}</td>
        <td>
          <Button color="success">Location</Button>
        </td>
      </tr>
    );
  }
}

BusResultRow.propTypes = {
  info: PropTypes.object
};

export default BusResultRow;
