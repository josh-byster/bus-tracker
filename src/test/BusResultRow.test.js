import React from "react";
import expect from "expect";
import { shallow } from "enzyme";
import BusResultRow from "../components/BusResultRow";
import renderer from "react-test-renderer";
import departures from "./mock_departures";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Bus Result Row Component", () => {
  const elem = renderer.create(
    <BusResultRow info={departures[0]} toggleModal={function() {}} />
  );
  elem.getInstance().setState({ validRequest: false });

  it("renders the proper h4 with invalid request", () => {
    expect(elem.toJSON()).toMatchSnapshot();
  });
});
