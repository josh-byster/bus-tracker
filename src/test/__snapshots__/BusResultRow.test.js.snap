// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bus Result Row Component renders the proper h4 with invalid request 1`] = `
<tr
  style={
    Object {
      "backgroundColor": "#fcee1f",
      "color": "#000000",
    }
  }
>
  <td
    width="40%"
  >
    <b>
      100N Yellow
    </b>
  </td>
  <td>
    5m
  </td>
  <td>
    5:32:40
  </td>
  <td>
    <button
      aria-label={null}
      className="btn btn-success"
      onClick={[Function]}
      type="button"
    >
      Location
    </button>
  </td>
</tr>
`;
